src/code/z_kankyo.c

```diff
@@ -967,249 +980,246 @@ void Environment_Update(PlayState* play, EnvironmentContext* envCtx, LightContex
-                        envCtx->lightSettings.light2Dir[0] = -envCtx->lightSettings.light1Dir[0];
-                        envCtx->lightSettings.light2Dir[1] = -envCtx->lightSettings.light1Dir[1];
-                        envCtx->lightSettings.light2Dir[2] = -envCtx->lightSettings.light1Dir[2];
-
-                        for (j = 0; j < 3; j++) {
-                            // blend light1Color
-                            blend8[0] =
-                                LERP(lightSettingsList[sTimeBasedLightConfigs[envCtx->lightConfig][i].lightSetting]
-                                         .light1Color[j],
-                                     lightSettingsList[sTimeBasedLightConfigs[envCtx->lightConfig][i].nextLightSetting]
-                                         .light1Color[j],
-                                     timeChangeBlend);
-                            blend8[1] = LERP(
-                                lightSettingsList[sTimeBasedLightConfigs[envCtx->changeLightNextConfig][i].lightSetting]
-                                    .light1Color[j],
-                                lightSettingsList[sTimeBasedLightConfigs[envCtx->changeLightNextConfig][i]
-                                                      .nextLightSetting]
-                                    .light1Color[j],
-                                timeChangeBlend);
-                            *(envCtx->lightSettings.light1Color + j) = LERP(blend8[0], blend8[1], configChangeBlend);
-
-                            // blend light2Color
-                            blend8[0] =
-                                LERP(lightSettingsList[sTimeBasedLightConfigs[envCtx->lightConfig][i].lightSetting]
-                                         .light2Color[j],
-                                     lightSettingsList[sTimeBasedLightConfigs[envCtx->lightConfig][i].nextLightSetting]
-                                         .light2Color[j],
-                                     timeChangeBlend);
-                            blend8[1] = LERP(
-                                lightSettingsList[sTimeBasedLightConfigs[envCtx->changeLightNextConfig][i].lightSetting]
-                                    .light2Color[j],
-                                lightSettingsList[sTimeBasedLightConfigs[envCtx->changeLightNextConfig][i]
-                                                      .nextLightSetting]
-                                    .light2Color[j],
-                                timeChangeBlend);
-                            *(envCtx->lightSettings.light2Color + j) = LERP(blend8[0], blend8[1], configChangeBlend);
+        if (envCtx->lightSettingOverride == LIGHT_SETTING_OVERRIDE_FULL_CONTROL) {
+        } else if ((envCtx->lightMode == LIGHT_MODE_TIME) &&
+                   (envCtx->lightSettingOverride == LIGHT_SETTING_OVERRIDE_NONE)) {
+            for (i = 0; i < ARRAY_COUNT(sTimeBasedLightConfigs[envCtx->lightConfig]); i++) {
```

Comment:
maybe `// Do nothing; Skip updating lights based on time or light settings` in the empty space

Reply:
I would also not merge the if else into the else of the `if (envCtx->lightSettingOverride == LIGHT_SETTING_OVERRIDE_FULL_CONTROL) {`

Reply:
I think it kinda makes sense, its all 3 ways the lights can get updated (even if the first case happens to be no updating at all)

Reply:
alright

Reply:
FWIW I like merging too, this function has way too much nesting already